float32 point1
float32 point2